sdk/contentsafety/azure-ai-contentsafety/src/main/java/com/azure/ai/contentsafety/ContentSafetyClient.java

```diff
@@ -440,221 +163,102 @@ public AnalyzeTextResult analyzeText(AnalyzeTextOptions body) {
-     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
-     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return all blockItems in a text blocklist as paginated response with {@link PagedIterable}.
-     */
-    @Generated
-    @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PagedIterable<TextBlockItem> listTextBlocklistItems(String blocklistName, Integer top, Integer skip) {
-        // Generated convenience method for listTextBlocklistItems
+    public AnalyzeImageResult analyzeImage(BinaryData content) {
+        // Customized convenience method for analyzeImageWithResponse
         RequestOptions requestOptions = new RequestOptions();
-        if (top != null) {
-            requestOptions.addQueryParam("top", String.valueOf(top), false);
-        }
-        if (skip != null) {
-            requestOptions.addQueryParam("skip", String.valueOf(skip), false);
-        }
-        return serviceClient
-                .listTextBlocklistItems(blocklistName, requestOptions)
-                .mapPage(bodyItemValue -> bodyItemValue.toObject(TextBlockItem.class));
-    }
-
-    /**
-     * Get All BlockItems By blocklistName
-     *
-     * <p>Get all blockItems in a text blocklist.
-     *
-     * @param blocklistName Text blocklist name.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws HttpResponseException thrown if the request is rejected by server.
-     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
-     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
-     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return all blockItems in a text blocklist as paginated response with {@link PagedIterable}.
-     */
-    @Generated
-    @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PagedIterable<TextBlockItem> listTextBlocklistItems(String blocklistName) {
-        // Generated convenience method for listTextBlocklistItems
-        RequestOptions requestOptions = new RequestOptions();
-        return serviceClient
-                .listTextBlocklistItems(blocklistName, requestOptions)
-                .mapPage(bodyItemValue -> bodyItemValue.toObject(TextBlockItem.class));
+        AnalyzeImageOptions body = new AnalyzeImageOptions(new ContentSafetyImageData().setContent(content));
+        return analyzeImageWithResponse(BinaryData.fromObject(body), requestOptions)
+            .getValue()
+            .toObject(AnalyzeImageResult.class);
```

Comment:
nit, same here, it usually better to call `analyzeImage`

Please check Async client as well.

Reply:
updated